src/librustc_typeck/error_codes.rs

```diff
@@ -3793,6 +3793,28 @@ details.
 [issue #33685]: https://github.com/rust-lang/rust/issues/33685
 "##,
 
+E0592: r##"
+This error occurs when you defined methods or associated functions with same
+name.
+
+For example, in the following code:
+
+```compile_fail,E0592
+struct Foo;
+
+impl Foo {
+    fn bar() {}
+}
+
+impl Foo {
+    fn bar() {}
+}
+```
+
+A similar error is E0201. The difference is whether there is one declaration
+block or not. To avoid this error, you have to give them one name each.
+"##,
```

Comment:
Can you add the previous failing example fixed (with comments in the code) please?

Reply:
I added fixed example (and comments to failing example).